custom_components/waste_collection_schedule/waste_collection_schedule/source/zys_harmonogram_pl.py

```diff
@@ -0,0 +1,123 @@
+    4: "mdi:recycle",
+    5: "mdi:recycle",
+    6: "mdi:recycle",
+    7: "mdi:trash-can",
+}
+
+
+class Source:
+    def __init__(self, city, street_name, street_number, commune_name):
+        self._city = city.upper()
+        self._street_name = street_name.upper()
+        self._street_number = street_number.upper()
+        self._commune_name = commune_name.lower()
+
+
+    def fetch(self):
+        try:
+            return self.get_data(API_URL.format(self._commune_name,datetime.datetime.now().year))
+        except Exception:
+            _LOGGER.debug(
+                f"fetch failed for source {TITLE}: trying different API_URL ..."
+            )
+            return self.get_data(API_URL.format(""))
+
+    def get_data(self, api_url):
+        r = requests.get(f"{api_url}/addresses/cities")
+        r.raise_for_status()
+        city_id = 0
+        cities = json.loads(r.text)
+        for item in cities:
+            if item["value"] == self._city:
+                city_id = item["id"]
+        if city_id == 0:
+            raise Exception("city not found")
+
+        r = requests.get(f"{api_url}/addresses/streets/{city_id}")
+        r.raise_for_status()
+        street_id = 0
+        streets = json.loads(r.text)
+        for item in streets:
+            if item["value"] == self._street_name:
+                street_id = item["id"]
+        if street_id == 0:
+            raise Exception("street not found")
+
+        r = requests.get(f"{api_url}/addresses/numbers/{city_id}/{street_id}")
+        r.raise_for_status()
+        number_id = 0
+        numbers = json.loads(r.text)
+        for item in numbers:
+            if item["value"] == self._street_number:
+                number_id = item["id"]
+        if number_id == 0:
+            raise Exception("number not found")
+
+        r = requests.get(f"{api_url}/reports?type=html&id={number_id}")
+        r.raise_for_status()
+        report = json.loads(r.text)
+        if report["status"] != "success":
+            raise Exception("fetch report failed")
+
+        r = requests.get(report["filePath"])
+        r.raise_for_status()
+        table = r.text[r.text.find("<table") : r.text.rfind("</table>") + 8]
+        tree = xml.etree.ElementTree.fromstring(table)
+        year = datetime.date.today().year
+
+        entries = []
+        for row_index, row in enumerate(tree.findall(".//tr")):
+            if row_index > 0:
+                for cell_index, cell in enumerate(row.findall(".//td")):
+                    if cell_index > 0 and isinstance(cell.text, str):
+                        for day in cell.text.split(","):
+                            entries.append(
+                                Collection(
+                                    datetime.date(year, row_index - 1, int(day)),
+                                    NAME_MAP[cell_index],
+                                    ICON_MAP[cell_index],
+                                )
+                            )
```

Comment:
I'd highly prefer to prevent this deep nesting by inverting your if condition and continue (if row_index == 0): continue

Is Hard-coding the index to names map a good idea? Wouldn't it be better to just read the name from the th's of the first row?